openmc/data/multipole.py

```diff
@@ -0,0 +1,535 @@
+                     'Possibly one more for a fission residue.')
+            if not np.issubdtype(data.dtype, np.complexfloating):
+                raise TypeError('Multipole data arrays must be complex dtype')
+        self._data = data
+
+    @windows.setter
+    def windows(self, windows):
+        if windows is not None:
+            cv.check_type('windows', windows, np.ndarray)
+            if len(windows.shape) != 2:
+                raise ValueError('Multipole windows arrays must be 2D')
+            if not np.issubdtype(windows.dtype, np.integer):
+                raise TypeError('Multipole windows arrays must be integer'
+                                ' dtype')
+        self._windows = windows
+
+    @broaden_poly.setter
+    def broaden_poly(self, broaden_poly):
+        if broaden_poly is not None:
+            cv.check_type('broaden_poly', broaden_poly, np.ndarray)
+            if len(broaden_poly.shape) != 1:
+                raise ValueError('Multipole broaden_poly arrays must be 1D')
+            if not np.issubdtype(broaden_poly.dtype, np.bool_):
+                raise TypeError('Multipole broaden_poly arrays must be boolean'
+                                ' dtype')
+        self._broaden_poly = broaden_poly
+
+    @curvefit.setter
+    def curvefit(self, curvefit):
+        if curvefit is not None:
+            cv.check_type('curvefit', curvefit, np.ndarray)
+            if len(curvefit.shape) != 3:
+                raise ValueError('Multipole curvefit arrays must be 3D')
+            if curvefit.shape[2] not in (2, 3):  # sig_s, sig_a (maybe sig_f)
+                raise ValueError('The third dimension of multipole curvefit'
+                                 ' arrays must have a length of 2 or 3')
+            if not np.issubdtype(curvefit.dtype, np.floating):
+                raise TypeError('Multipole curvefit arrays must be float dtype')
+        self._curvefit = curvefit
+
+    @classmethod
+    def from_hdf5(cls, group_or_filename):
+        """Construct a WindowedMultipole object from an HDF5 group or file.
+
+        Parameters
+        ----------
+        group_or_filename : h5py.Group or str
+            HDF5 group containing multipole data. If given as a string, it is
+            assumed to be the filename for the HDF5 file, and the first group is
+            used to read from.
+
+        Returns
+        -------
+        openmc.data.WindowedMultipole
+            Resonant cross sections represented in the windowed multipole
+            format.
+
+        """
+
+        if isinstance(group_or_filename, h5py.Group):
+            group = group_or_filename
+        else:
+            h5file = h5py.File(str(group_or_filename), 'r')
+
+            # Make sure version matches
+            if 'version' in h5file.attrs:
+                major, minor = h5file.attrs['version']
+                if major != WMP_VERSION_MAJOR:
+                    raise IOError(
```

Comment:
I would use a better exception to describe the situation such as a `ModuleNotFoundException` https://docs.python.org/3/library/exceptions.html#ModuleNotFoundError